File: Leetcode/Algorithms/Medium/MaxNumberOfKSumPairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        nums.sort()
        count = 0
        start = 0
        end = len(nums) - 1
        while(start < end):
            val = nums[start] + nums[end]
            if(val == k):
                count += 1
                nums.pop(start)
                end -= 1
                nums.pop(end)
                end -= 1
            elif(val > k):
                end-=1
            elif(val < k):
                start+=1
            
        return count

File: Leetcode/Algorithms/Medium/MaxNumberOfKSumPairs/MaxNumberOfKSumPairs.cpp
class Solution
{
public:
    int maxOperations(vector<int> &nums, int k)
    {
        sort(nums.begin(), nums.end());

        int start = 0;
        int end = nums.size() - 1;
        int count = 0;
        while (start < end)
        {
            int val = nums[start] + nums[end];
            if (val > k)
                end--;
            else if (val < k)
                start++;
            else if (val == k)
            {
                start++;
                end--;
                count++;
            }
        }
        return count;
    }
};

// Time Complexity - O(NlogN)
// Space Complexity - O(1)